src/python/impactx/plot/Survey.py

```diff
@@ -0,0 +1,129 @@
+"""
+This file is part of ImpactX
+
+Copyright 2025 ImpactX contributors
+Authors: Axel Huebl
+License: BSD-3-Clause-LBNL
+"""
+
+
+def plot_survey(
+    self, ref=None, ax=None, legend=True, legend_ncols=5, palette="cern-lhc"
+):
+    """Plot over s of all elements in the KnownElementsList.
+
+    The signs of element strengths are determined by the sign of the charge of the reference particle.
+    The projection of all element strengths is s-x ("vertical").
+
+    Parameters
+    ----------
+    self : ImpactXParticleContainer_*
+        The KnownElementsList class in ImpactX
+    ref : RefPart
+        A reference particle, checked for the charge sign to plot focusing/defocusing strength directions properly.
+    ax : matplotlib axes
+        A plotting area in matplotlib (called axes there).
+    legend: bool
+        Plot a legend if true.
+    legend_ncols: int
+        Number of columns for lattice element types in the legend.
+    palette: string
+        Color palette.
+
+    Returns
+    -------
+    Either populates the matplotlib axes in ax or creates a new axes containing the plot.
+    """
+    from math import copysign
+
+    import matplotlib.pyplot as plt
+    import numpy as np
+    from matplotlib.patches import Rectangle
+
+    from .ElementColors import get_element_color
+
+    charge_qe = 1.0 if ref is None else ref.charge_qe
+
+    ax = ax or plt.subplot(111)
+
+    element_lengths = [element.ds for element in self]
+    element_s = np.cumulative_sum(element_lengths, include_initial=True)
```

Comment:
Oops, this requires NumPy 2.1+

Reply:
Ah dang, NumPy 2.1+ only supports Python 3.10+, but 3.9 is not yet EOL, so a bit rough to ditch.

Reply:
work-around added